diff --git a/krb5/src/lib/krb5/ccache/ccbase.c b/krb5/src/lib/krb5/ccache/ccbase.c
--- a/krb5/src/lib/krb5/ccache/ccbase.c
+++ b/krb5/src/lib/krb5/ccache/ccbase.c
@@ -198,2 +198,2 @@
-    char *pfx, *cp;
-    const char *resid;
+    char *pfx;
+    const char *cp, *resid;
diff --git a/krb5/src/lib/krb5/os/expand_path.c b/krb5/src/lib/krb5/os/expand_path.c
--- a/krb5/src/lib/krb5/os/expand_path.c
+++ b/krb5/src/lib/krb5/os/expand_path.c
@@ -457,2 +457,2 @@
-    char *tok_begin, *tok_end, *tok_val, **extra_tokens = NULL;
-    const char *path_left;
+    char *tok_val, **extra_tokens = NULL;
+    const char *tok_begin, *tok_end, *path_left;
diff --git a/krb5/src/lib/krb5/os/locate_kdc.c b/krb5/src/lib/krb5/os/locate_kdc.c
--- a/krb5/src/lib/krb5/os/locate_kdc.c
+++ b/krb5/src/lib/krb5/os/locate_kdc.c
@@ -191,2 +191,2 @@
-parse_uri_if_https(const char *host_or_uri, k5_transport *transport,
-                   const char **host, const char **uri_path)
+parse_uri_if_https(char *host_or_uri, k5_transport *transport,
+                   char **host, const char **uri_path)
@@ -232,2 +232,1 @@
-    char **hostlist = NULL, *realmstr = NULL, *host = NULL;
-    const char *hostspec;
+    char **hostlist = NULL, *realmstr = NULL, *host = NULL, *hostspec;
@@ -538,2 +537,2 @@
-parse_uri_fields(const char *uri, k5_transport *transport_out,
-                 const char **host_out, int *primary_out)
+parse_uri_fields(char *uri, k5_transport *transport_out,
+                 char **host_out, int *primary_out)
@@ -607,2 +606,2 @@
-    char *host;
-    const char *host_field, *path;
+    char *host, *host_field;
+    const char *path;
