diff --git a/modules/caddyhttp/celmatcher.go b/modules/caddyhttp/celmatcher.go
index 4c79bf2d..c450fc99 100644
--- a/modules/caddyhttp/celmatcher.go
+++ b/modules/caddyhttp/celmatcher.go
@@ -506 +506 @@ func CELMatcherDecorator(funcName string, fac any) interpreter.InterpretableDeco
-				[]interpreter.Interpretable{reqAttr},
+				[]interpreter.InterpretableV2{reqAttr},
@@ -529 +529 @@ func CELMatcherDecorator(funcName string, fac any) interpreter.InterpretableDeco
-				[]interpreter.Interpretable{reqAttr},
+				[]interpreter.InterpretableV2{reqAttr},
